app/services/category_hashtag_data_source.rb

```diff
@@ -52,11 +52,43 @@ def self.search(guardian, term, limit)
   end
 
   def self.search_sort(search_results, term)
-    search_results
-      .select { |item| item.slug == term }
-      .sort_by { |item| item.text.downcase }
-      .concat(
-        search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
-      )
+    if term.present?
+      search_results
+        .select { |item| item.slug == term }
+        .sort_by { |item| item.text.downcase }
+        .concat(
+          search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
+        )
```

Comment:
```suggestion
      search_results.sort_by do |item|
        [item.slug == term ? 0 : 1, item.text.downcase]
      end
```

I think we can simplify this to a single loop.

Reply:
I did try so many variations of this sorting and it still ended up wrong...but I did not try this one so I will see if it will work :+1:

Reply:
Ah I see the difference now, I didn't know that you could return multiple values in an array for `sort_by`, this works!